cmake/CPM.cmake

```diff
@@ -134,6 +134,17 @@ endif()
 include(FetchContent)
 include(CMakeParseArguments)
 
+# Infer package name from git repository uri (path or url)
+function(cpm_package_name_from_git_uri URI RESULT)
+  string(REGEX MATCH "([^/:]+)/?.git/?$" cpmGitUriMatch "${URI}")
+  if(DEFINED cpmGitUriMatch)
+    set(${RESULT}
+        ${CMAKE_MATCH_1}
+        PARENT_SCOPE
+    )
+  endif()
```

Comment:
It may be a good idea to unset `${RESULT}` in the parent scope when no match is found. Otherwise the value could be mistaken with an unrelated outside variable using the same name.